include/eve/arch/cpu/wide.hpp

```diff
@@ -70,6 +70,14 @@ namespace eve
   {
     using storage_base = detail::wide_storage<as_register_t<Type, Cardinal, abi_t<Type, Cardinal>>>;
 
+    // split_type does not exist on cardinals of size 1
+    // this function returns a void_t if the split_type does not exist
+    template <typename T, typename N>
+    static constexpr auto get_split_type() noexcept
+    {
+      if constexpr (N::value > 1) return wide<T, typename N::split_type>{};
+    }
```

Comment:
Why is this function needed? I am not wild about it in the public api.
Also - it's just void, not void_t

Reply:
It is a public function on wide. If you don't want it to be public - make it private or put in detail.